concent_api/core/validation.py

```diff
@@ -384,3 +386,27 @@ def validate_uuid(id_: str) -> None:
             'ID must be a UUID derivative.',
             error_code=ErrorCode.MESSAGE_WRONG_UUID_VALUE,
         )
+
+
+def validate_database_task_to_compute(
+        task_to_compute: message.tasks.TaskToCompute,
+        message_to_compare: message.Message
+) -> None:
+    if message_to_compare.task_to_compute != task_to_compute:
+        raise ValidationError({
+            message_to_compare.__class__.__name__: (
+                'Nested TaskToCompute message must be the same as TaskToCompute stored separately'
+            )
+        })
+
+
+def validate_database_report_computed_task(
+        report_computed_task: message.tasks.ReportComputedTask,
+        message_to_compare: message.Message
+) -> None:
+    if message_to_compare.report_computed_task != report_computed_task:
+        raise ValidationError({
+            message_to_compare.__class__.__name__: (
+                'Nested ReportComputedTask with TaskToCompute message must be the same as both messages stored separately'
```

Comment:
here, you are checking just `ReportComputedTask`, so error message should reflect that ;)